google/auth/impersonated_credentials.py

```diff
@@ -251,13 +259,15 @@ def __init__(
         self._quota_project_id = quota_project_id
         self._iam_endpoint_override = iam_endpoint_override
         self._cred_file_path = None
+        self._trust_boundary = trust_boundary
 
     def _metric_header_for_usage(self):
         return metrics.CRED_TYPE_SA_IMPERSONATE
 
     @_helpers.copy_docstring(credentials.Credentials)
     def refresh(self, request):
         self._update_token(request)
+        self._refresh_trust_boundary(request)
```

Comment:
One suggestion for a potential refactoring to improve the design:

Currently, each class that inherits from `CredentialsWithTrustBoundary` (e.g., `compute_engine.Credentials`, `impersonated_credentials.Credentials`, `service_account.Credentials`) is required to explicitly call `self._refresh_trust_boundary(request)` within its own `refresh` method.

To enhance encapsulation and reduce the burden on subclasses, it would be preferable if this call was handled automatically by the base class. Could this be refactored?